decrypt/decrypt.go

```diff
@@ -1,43 +1,248 @@
+	files := []helpers.EncryptionFileSet{}
+	for _, filename := range Args.Args() {
+
+		// Set directory for the output file
+		unencryptedFilename := strings.TrimSuffix(filename, ".c4gh")
+
+		files = append(files, helpers.EncryptionFileSet{Encrypted: filename, Unencrypted: unencryptedFilename})
+	}
+
+	// Create a private key and exit if the -createKey flag is set
+	// Currently, no password can be provided.
+	if *keyName != "" {
+		err := generateKeyPair(*keyName, "")
+		if err != nil {
+			return fmt.Errorf("failed to generate key pair: %s", err)
+		}
+		os.Exit(0)
+	}
 
 	// Check that we have a private key to decrypt with
-	if *privateKey == "" {
-		log.Fatal("A private key is required to decrypt data")
+	if *privateKeyFile == "" {
+		return errors.New("a private key is required to decrypt data")
+	}
+
+	// Loading private key file
+	privateKey, err := readPrivateKey(*privateKeyFile, "")
+	if err != nil {
+		return err
+	}
+
+	// Check that all the encrypted files exist, and all the unencrypted don't
+	err = checkFiles(files)
+	if err != nil {
+		return err
+	}
+
+	// decrypt the input files
+	numFiles := len(files)
+	for i, file := range files {
+		log.Infof("Decrypting file %v/%v: %s", i+1, numFiles, file.Encrypted)
+
+		err = decrypt(file.Encrypted, file.Unencrypted, *privateKey)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Generates a crypt4gh key pair, and saves it to the `<basename>.pub.pem` and
+// `<basename>.sec.pem` files. If any of the files already exists, the function
+// will instead return an error.
+func generateKeyPair(basename, password string) error {
+	privateKeyName := fmt.Sprintf("%s.sec.pem", basename)
+	publicKeyName := fmt.Sprintf("%s.pub.pem", basename)
+
+	// check if any of the files exist
+	if helpers.FileExists(publicKeyName) {
+		return fmt.Errorf("public key file %s already exists", publicKeyName)
+	}
+	if helpers.FileExists(privateKeyName) {
+		return fmt.Errorf("private key file %s already exists", privateKeyName)
+	}
+
+	// Generate key pair
+	log.Infof("Generating key pair: %s, %s", privateKeyName, publicKeyName)
+
+	publicKeyData, privateKeyData, err := keys.GenerateKeyPair()
+	if err != nil {
+		return err
+	}
+
+	// Save keys to file
+	pubFile, err := os.OpenFile(filepath.Clean(publicKeyName), os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := pubFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+	err = keys.WriteCrypt4GHX25519PublicKey(pubFile, publicKeyData)
+	if err != nil {
+		return err
+	}
+
+	secFile, err := os.OpenFile(filepath.Clean(privateKeyName), os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := secFile.Close(); err != nil {
+			log.Errorf("Error closing file: %s\n", err)
+		}
+	}()
+	pass := []byte(password)
+	err = keys.WriteCrypt4GHX25519PrivateKey(secFile, privateKeyData, pass)
+
+	return err
+}
```

Comment:
```suggestion
```

this is probably a copy paste error, since by now you should already have your keypair.

Reply:
yes :D